### ISSUE TITLE
'NoneType' object has no attribute 'name' when editing an expired override with the CLI

### ISSUE THREAD

**Issue by bowlofeggs** [fedora-infra/bodhi#1626]
I received the following traceback when trying to edit an expired override (to extend/unexpire it):

```
$ bodhi overrides edit python-fedora-0.9.0-3.el7 --duration 2 --notes "Needed for Bodhi 2.8.0."
Traceback (most recent call last):
  File "/usr/bin/bodhi", line 11, in <module>
    load_entry_point('bodhi-client==2.8.0b1', 'console_scripts', 'bodhi')()
  File "/usr/lib/python2.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python2.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python2.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python2.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python2.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/bodhi/client/__init__.py", line 477, in edit_buildroot_overrides
    _save_override(url=url, user=user, password=password, staging=staging, edit=True, **kwargs)
  File "/usr/lib/python2.7/site-packages/bodhi/client/__init__.py", line 121, in _save_override
    expired=kwargs.get('expire', False))
  File "/usr/lib/python2.7/site-packages/bodhi/client/bindings.py", line 97, in wrapper
    raise BodhiClientException(problems)
bodhi.client.bindings.BodhiClientException: Unable to save buildroot override: 'NoneType' object has no attribute 'name'
```

The server also had a traceback, which revealed that I had not authenticated:

```
2017-06-19 18:45:58,003 ERROR [bodhi.server][Dummy-5] 'NoneType' object has no attribute 'name'
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/bodhi/server/services/overrides.py", line 186, in save_override
    submitter = User.get(request.user.name, request.db)
AttributeError: 'NoneType' object has no attribute 'name'
```

This may be connected to and may be fixed by https://github.com/fedora-infra/bodhi/pull/1566. Thus, there may be nothing to do, but I wanted to file this ticket so we could follow up and ensure that there isn't still a problem.

For the record, I am able to reproduce this problem with CLI versions 2.7.0 and 2.8.0 against server version 2.7.0 (production). I have no attempted to test against server version 2.8.0 or the develop branch yet.

**Comment by ryanlerch** [comment]
This appears to be fixed in develop client / and server. Just confirmed that i got prompted for my username and password when trying to re-open an expired override.

**Comment by bowlofeggs** [comment]
The patch for this issue should be included in the upcoming 2.10.0 release:

https://github.com/fedora-infra/bodhi/pull/1744
